frontend/cypress/integration/common/graph_toolbar.ts

```diff
@@ -131,18 +132,35 @@ Then(`user does not see any traffic`, () => {
 Then('user sees graph duration menu', () => {
   cy.get('button#time_range_duration-toggle').invoke('attr', 'aria-expanded').should('eq', 'true');
   cy.get('ul#time_range_duration').within(() => {
-    cy.get('li').should('have.length', 11);
-    cy.get('li#60').should('exist');
-    cy.get('li#120').should('exist');
-    cy.get('li#300').should('exist');
-    cy.get('li#600').should('exist');
-    cy.get('li#1800').should('exist');
-    cy.get('li#3600').should('exist');
-    cy.get('li#10800').should('exist');
-    cy.get('li#21600').should('exist');
-    cy.get('li#43200').should('exist');
-    cy.get('li#86400').should('exist');
-    cy.get('li#604800').should('exist');
+    cy.get('li')
+      .should('have.length.within', 7, 11)
+      .then(elems => {
+        // number of entries can vary based on the test environment. prometheus may be configured with
+        //   scrapeInterval: 15s, 1m
+        //   dataRetention: 6h, 15d
+        const durations = elems.length;
+
+        if (durations !== 7 && durations !== 8 && durations !== 11) {
+          fail(`Unexpected prom config in test environment, check scrapeInterval and/or dataRetentionTime`);
+        }
+        // check the durations we always expect
+        cy.get('li#120').should('exist');
+        cy.get('li#300').should('exist');
+        cy.get('li#600').should('exist');
+        cy.get('li#1800').should('exist');
+        cy.get('li#3600').should('exist');
+        cy.get('li#10800').should('exist');
+        cy.get('li#21600').should('exist');
+
+        if (durations === 8 || durations === 11) {
+          cy.get('li#60').should('exist');
+        }
+        if (durations >= 10) {
+          cy.get('li#43200').should('exist');
+          cy.get('li#86400').should('exist');
+          cy.get('li#604800').should('exist');
+        }
```

Comment:
I tried getting the ServerConfig but I couldn't get the cypress tests to be able to access it.  If you know a way I'm happy to change back to that approach.  Since I couldn't figure a way to do it I reverted to this sort of "soft" approach.

Reply:
I guess maybe my problem was trying to grab it via the UI code, but I guess we could make an API call to get the config?

Reply:
> I guess maybe my problem was trying to grab it via the UI code, but I guess we could make an API call to get the config?

Yeah you can do something like `cy.request('api/config').then(...)` and that should get you the retention period.

Reply:
I learned something today.